### PR TITLE
feat(GRO-1117): Add follow galleries I love screen

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [GRO-1117]

### Description

- [x] User sees follow galleries screen
- [x] User sees default list of galleries
- [x] User can search galleries
- [x] User sees gallery names populating after search
- [x] User can continue after selecting at least one gallery
- [x] User can follow as many galleries as they would like 
- [ ] user can click skip and leave the flow and return to homescreen
- [ ] user can click back
- [ ] next takes you to the homescreen 

the navigation steps will be addressed in another pr.

cc @artsy/grow-devs 

#### Demo Video

https://user-images.githubusercontent.com/21178754/181765972-ab388de5-6fd9-496e-b46b-77b1810de151.mp4

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [x] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add follow galleries I love screen new onboarding screen - gkartalis

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[GRO-1117]: https://artsyproduct.atlassian.net/browse/GRO-1117?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Onboarding/OnboardingPersonalization2/OnboardingOrderedSet.tsx [hunk 1]
```diff
@@ -52,6 +54,22 @@ const OnboardingOrderedSet: React.FC<OnboardingOrderedSetProps> = ({ id }) => {
                 py={space(1)}
               />
             )
+          case "Profile": {
+            const partner = item.owner
+
+            if (!partner || partner.__typename !== "Partner") {
```
Comment (olerichter00) [inline src/app/Scenes/Onboarding/OnboardingPersonalization2/OnboardingOrderedSet.tsx:60]: Does `__typename` actually work anymore? There are a few breaking changes in [Relay v14.0.0](https://github.com/facebook/relay/releases/tag/v14.0.0) that disallow it on queries and I remember I had to fix it in one place after the upgrade 🤔

Comment (gkartalis) [reply src/app/Scenes/Onboarding/OnboardingPersonalization2/OnboardingOrderedSet.tsx:60]: it seems like it works without any issue 🤔

Comment (olerichter00) [reply src/app/Scenes/Onboarding/OnboardingPersonalization2/OnboardingOrderedSet.tsx:60]: That's great 👍

Comment (damassi) [reply src/app/Scenes/Onboarding/OnboardingPersonalization2/OnboardingOrderedSet.tsx:60]: That would have been very upsetting if they removed `__typename`!

Comment (damassi) [reply src/app/Scenes/Onboarding/OnboardingPersonalization2/OnboardingOrderedSet.tsx:60]: (And i can't believe y'all are on on Relay 14!)

**Review by damassi — DISMISSED** [review]
👌

**Review by olerichter00 — APPROVED** [review]

**Pull request merged**